tests/AutoReview/ProjectCodeTest.php

```diff
@@ -452,6 +452,56 @@ public function testExpectedInputOrder(string $testClassName): void
         }
     }
 
+    /**
+     * @dataProvider provideTestClassCases
+     */
+    public function testDataPorvidersDeclaredReturnType(string $testClassName): void
```

Comment:
```suggestion
    public function testDataProvidersDeclaredReturnType(string $testClassName): void
```

Anyway, why do we need this test? Isn't PHPStan checking this?

Reply:
looking on amount of changes, PHPStan is not checking it.

Reply:
yep, I wanted to avoid `array<int>` when we have only one int, so `array{int}`.
some side wins on the go also, like return array vs type vs iterable unification.
now (With this PR), for data providers, we either have no phpdoc (and simply rely on `: iterable`), or we have nice iterable of tuples